### DISCUSSION TITLE
Is there a way to learn which keys a game uses?

### DISCUSSION THREAD

**Discussion by lwcorp** [joncampbell123/dosbox-x#4200] [Q&A]
Some games have terrible manuals or no manuals at all. Is there any way to click various keyboard buttons and use DOSBox-X to see which of them reacts to presses inside the game? Like showing which keys the game listens to.

**Comment by Torinde** [comment]
Finding the keys is a game in itself..,
Imagine running a random .exe (without any manuals or even an idea what it does) in a language you don't know, and trying to deduce the game mechanics by trial and error... [like AI learning to play Atari games...]
You end up getting to the last boss in an "unconsciously handicaped mode" (e.g. without using the "secondary power" of your weapon because you didn't realize it exists)... and you play that level for ages, in frustration.... because it's unbeatable without that "simple trick" of just using a normal function of the game.

Years later, after the internet, you read about it and "aha!" - then the whole game is much easier... but much less challenging/interesting... all those special solutions for earlier levels you found - not required anymore, because with the secondary power you just walk trough the barriers :)

TLDR: I'm not sure DOSbox can detect what a game listens to - and if it can't - the challenge of manual discovery may be enjoyable :)

**Comment by lwcorp** [comment]
"Like AI learning" is exactly what I want, but those AIs so far refuse to play even online games...

If you find the challenge enjoyable, try the game _WWF European Rampage Tour_ (which can also be played with sites that use an online DOSBox). The challenges by priority are:
1. In practice mode, find out how to switch the second team into another team (based on memories online that it can be done).
2. Find any cheat mode (Amiga version has it and the PC version is supposed to be based on it).
3. Find out how to exit the game (just because it's annoying to think otherwise in the original runs one had to restart in order to exit).
